src/vs/editor/contrib/hover/browser/getHover.ts

```diff
@@ -33,19 +33,21 @@ async function executeProvider(provider: HoverProvider, ordinal: number, model:
 	return undefined;
 }
 
-export function getHover(registry: LanguageFeatureRegistry<HoverProvider>, model: ITextModel, position: Position, token: CancellationToken): AsyncIterableObject<HoverProviderResult> {
+export function getHover(registry: LanguageFeatureRegistry<HoverProvider>, model: ITextModel, position: Position, context: HoverContext | undefined, token: CancellationToken): AsyncIterableObject<HoverProviderResult> {
 	const providers = registry.ordered(model);
-	const promises = providers.map((provider, index) => executeProvider(provider, index, model, position, token));
+	const promises = providers.map((provider, index) => executeProvider(provider, index, model, position, context, token));
 	return AsyncIterableObject.fromPromises(promises).coalesce();
 }
 
-export function getHoverPromise(registry: LanguageFeatureRegistry<HoverProvider>, model: ITextModel, position: Position, token: CancellationToken): Promise<Hover[]> {
-	return getHover(registry, model, position, token).map(item => item.hover).toPromise();
+export function getHoverPromise(registry: LanguageFeatureRegistry<HoverProvider>, model: ITextModel, position: Position, context: HoverContext | undefined, token: CancellationToken): Promise<Hover[]> {
+	return getHover(registry, model, position, context, token).map(item => item.hover).toPromise();
 }
 
-registerModelAndPositionCommand('_executeHoverProvider', (accessor, model, position) => {
+registerModelAndPositionCommand('_executeHoverProvider', async (accessor, model, position, context) => {
 	const languageFeaturesService = accessor.get(ILanguageFeaturesService);
-	return getHoverPromise(languageFeaturesService.hoverProvider, model, position, CancellationToken.None);
+	const hovers = await getHoverPromise(languageFeaturesService.hoverProvider, model, position, context, CancellationToken.None);
+	hovers.forEach(hover => hover.dispose());
```

Comment:
This is fishy, but okay for now.
A disposed object really shouldn't be used anymore.
Maybe can you copy the hovers into new objects before you dispose the original?
```
const result = hovers.map(h => cloneHover(h));
hovers.forEach(hover => hover.dispose());
return result;
```

Reply:
good idea